opensrp-chw/src/main/java/org/smartregister/chw/activity/AncMemberMapActivity.java

```diff
@@ -0,0 +1,259 @@
+                        communityTransportersSource = style.getSourceAs("community-transporters-data-set");
+
+                        FeatureCollection featureCollection = loadCommunityTransporters();
+                        BoundingBox boundingBox = showCommunityTransporters(mapboxMap, featureCollection);
+
+                        zoomToPatientLocation(mapboxMap, boundingBox);
+                        addCommunityTransporterClickListener(kujakuMapView);
+                    }
+                });
+            }
+        });
+    }
+
+    private void addCommunityTransporterClickListener(@NonNull KujakuMapView kujakuMapView) {
+        kujakuMapView.setOnFeatureClickListener(new OnFeatureClickListener() {
+            @Override
+            public void onFeatureClick(List<Feature> features) {
+                // We only pick the first one
+                Feature feature = features.get(0);
+
+                Number recyclerViewPosition = feature.getNumberProperty(RECYCLER_VIEW_POSITION_PROPERTY);
+                if (recyclerViewPosition != null) {
+                    scrollToCardAtPosition(recyclerViewPosition.intValue());
+                }
+
+                featureClicked(feature);
+            }
+        }, "community-transporters");
+    }
+
+    private void featureClicked(@NonNull Feature feature) {
+        String responderName = feature.getStringProperty(CoreConstants.JsonAssets.RESPONDER_NAME);
+        String respondersPhoneNumber = feature.getStringProperty(CoreConstants.JsonAssets.RESPONDER_PHONE_NUMBER);
+        Toast.makeText(AncMemberMapActivity.this, String.format("Clicked on RESPONDER\nName: %s\nPhone No: %s", responderName, respondersPhoneNumber), Toast.LENGTH_LONG)
+                .show();
+    }
+
+    private void scrollToCardAtPosition(int itemPosition) {
+
+    }
+
+    @Nullable
+    private LatLng extractUserLocation(Bundle savedInstanceState) {
+        double latitude = Double.parseDouble("-1.9885");
+        double longitude = Double.parseDouble("33.7799");
+        return new LatLng(latitude, longitude);
+    }
+
+    private void zoomToPatientLocation(@NonNull MapboxMap mapboxMap, @Nullable BoundingBox boundingBox) {
+        if (userLocation != null && boundingBox == null) {
+            CameraPosition cameraPosition = new CameraPosition.Builder()
+                    .target(userLocation)
+                    .zoom(16)
+                    .build();
+
+            mapboxMap.animateCamera(CameraUpdateFactory.newCameraPosition(cameraPosition));
+        }
+
+        if (userLocation != null) {
+            MarkerOptions markerOptions = markerOptions = new MarkerOptions()
+                    .position(userLocation);
+            mapboxMap.addMarker(markerOptions);
+        }
+
+        if (boundingBox != null) {
+            if (!CoordinateUtils.isLocationInBounds(userLocation, boundingBox.north(), boundingBox.south(), boundingBox.east(), boundingBox.west())) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5493232)